lib/spack/docs/index.rst

```diff
@@ -49,6 +49,12 @@ or refer to the full manual below.
 
 .. toctree::
    :maxdepth: 2
+   :caption: Module File Support
+
+   module_file_support
```

Comment:
I agree with @adamjstewart... this should not be a top-level tree.  It should go in the reference manual or tutorial --- or split between the two.